src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArr = [];
+  let copyState = { ...state };
+
+  const ACTION_CASES = {
+    add: 'addProperties',
+    remove: 'removeProperties',
+    clear: 'clear',
+  };
+
+  for (let i = 0; i < actions.length; i++) {
+    const ACTION = actions[i];
```

Comment:
it would be more readable and clear if you use `for...of ` loop and a destructuring like:

```
 for (const action of actions) {
    const { type, extraData, keysToRemove } = action;
```